### PR TITLE
Rework `opensea_v3_polygon.events`

### PR DESCRIPTION
**Pull request by sohwak**
Brief comments on the purpose of your changes:

Additional needs : 
- Column names for `seaport_polygon.Seaport` may change. I will modify the script accordingly after the change.

Fixes :
- In current script for Seaport v1.1, purchases made with `matchAdvancedOrders` were counted twice in the existing source. 
I removed the `matchAdvancedOrders` and `matchOrders` parts from the existing source by `receipient != '0x00000000000000000000000000000000000000000000'` and added CTE parts with function tables.
- non-OpenSea trades would be filtered with zone address and Opensea fee_wallet.
- More accurate in platform fee / creator earning.

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by 0xRobin** [comment]
@sohwak would this make the polygon v3 models in line with the ethereum v3 models?
Is this change already present in the other opensea v3 models?

**Comment by sohwak** [comment]
> @sohwak would this make the polygon v3 models in line with the ethereum v3 models?
> Is this change already present in the other opensea v3 models?

No. I'm planning to apply this model to Ethereum v3 in near soon.

**Comment by 0xRobin** [comment]
> No. I'm planning to apply this model to Ethereum v3 in near soon.

Ok that's fine. 
While diverging model logic (between diff chains) is a bit of a concern, I'm fine following you're approach here and taking it 1 step at a time. :+1: 
In the future I can help transform these to use macro's to define the implementation logic, each chain specific model can then use the same macro (and thus the same logic)

**Comment by sohwak** [comment]
Thanks @0xRobin , if there's a good way to do it, I'd love to try it. 😀

This PR is currently waiting for a column name change in the decoding tables. 
https://discord.com/channels/757637422384283659/1111275327365316658

I was going to change the source when the column name is changed, so I think it's better to wait until that is done.

**Review by 0xRobin — APPROVED** [review]

**Comment by sohwak** [comment]
Hi @jeff-dude, Could you check this and merge it?

File: models/opensea/polygon/opensea_v3_polygon_events.sql [hunk 1]
```diff
@@ -1,67 +1,729 @@
...
+{% set c_alternative_token_address = "0x0000000000000000000000000000000000001010" %}  -- MATIC
+{% set c_native_symbol = "MATIC" %}
+
+with source_polygon_transactions as (
+    select *
+      from {{ source('polygon','transactions') }}
+    {% if not is_incremental() %}
+     where block_time >= date '{{c_seaport_first_date}}'  -- seaport first txn
+    {% endif %}
+    {% if is_incremental() %}
```
Comment (jeff-dude) [inline models/opensea/polygon/opensea_v3_polygon_events.sql:33]: i won't block the merge on this, but curious if interval needs to be three weeks intentionally here?
cc @0xRobin for awareness when rebuilt in new standard

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**